spinta/types/model.py

```diff
@@ -47,6 +47,16 @@ def load(context: Context, prop: Property, data: dict, manifest: Manifest) -> Pr
     return prop
 
 
+@load.register()
+def load(context: Context, model: Model, data: dict) -> dict:
+    for name, prop in model.properties.items():
+        if name in data:
+            data_value = data[name]
+            # XXX: rewrite into command instead of type method?
```

Comment:
It is OK to have methods on component classes as long, as those methods stay within component domain. If you need to tho something more, that is out of component domain, then you need to use commands or functions.